estate/models/estate_property.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, models, fields, exceptions
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Estate Property"
+
+    name = fields.Char(required=True, default="Unknown")
+    description = fields.Text()
+    postcode = fields.Char()
+    date_avaliability = fields.Date(
+        copy=False, default=fields.Date.today() + relativedelta(month=3)
+    )
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    last_seen = fields.Datetime("Last Seen", default=fields.Datetime.now)
+    garden_orientation = fields.Selection(
+        selection=[
+            ("north", "North"),
+            ("south", "South"),
+            ("east", "East"),
+            ("west", "West"),
+        ],
+        string="Orientation",
+    )
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        selection=[
+            ("new", "New"),
+            ("offer_received", "Offer Received"),
+            ("offer_accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("cancelled", "Cancelled"),
+        ],
+        string="Status",
+        required=True,
+        copy=False,
+        default="new",
+    )
+    property_type_id = fields.Many2one("estate.property.type", string="Property Type")
+    customer = fields.Many2one("res.partner", string="Customer", copy=False)
+    salesperson = fields.Many2one(
+        "res.users", string="Salesperson", default=lambda self: self.env.user
+    )
+    tag_ids = fields.Many2many("estate.property.tag", string="Property Tags")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offer")
+    total_area = fields.Integer(compute="_compute_area")
+    best_price = fields.Integer(compute="_compute_highest")
```

Comment:
```suggestion
    best_price = fields.Integer(compute="_compute_best_price")
```
It is a good habit to compute the method name related to the field name. So others can easily identify.